ts_src/types.ts

```diff
@@ -65,6 +67,48 @@ export const Network = typeforce.compile({
   wif: typeforce.UInt8,
 });
 
+export interface XOnlyPointAddTweakResult {
+  parity: 1 | 0;
+  xOnlyPubkey: Uint8Array;
+}
+
+export interface Tapleaf {
+  output: Buffer;
+  version?: number;
+}
+
+export const TAPLEAF_VERSION_MASK = 0xfe;
+export function isTapleaf(o: any): o is Tapleaf {
+  if (!o || !('output' in o)) return false;
+  if (!NBuffer.isBuffer(o.output)) return false;
+  if (o.version !== undefined)
+    return (o.version & TAPLEAF_VERSION_MASK) === o.version;
+  return true;
+}
+
+/**
+ * Binary tree repsenting script path spends for a Taproot input.
+ * Each node is either a single Tapleaf, or a pair of Tapleaf | Taptree.
+ * The tree has no balancing requirements.
+ */
+export type Taptree = [Taptree | Tapleaf, Taptree | Tapleaf] | Tapleaf;
+
+export function isTaptree(scriptTree: any): scriptTree is Taptree {
+  if (!Array(scriptTree)) return isTapleaf(scriptTree);
```

Comment:
was this supposed to be `Array.isArray()`? This is calling the constructor and always returns a truthy value

Reply:
nevermind - we have a `const Array = ` further down